### PR TITLE
[ART-1858] update validation for promote job

### PR DESCRIPTION
**Pull request by Ximinhan**
along with this PR https://github.com/openshift/aos-cd-jobs/pull/2240/files

### CODE REVIEW

File: jobs/build/pre-release/Jenkinsfile [hunk 1]
```diff
@@ -136,7 +136,7 @@ node {
             def CLIENT_TYPE = "ocp-dev-preview"
 
             stage("validation") {
-                release.stageValidation(quay_url, dest_release_tag, -1, params.PERMIT_PAYLOAD_OVERWRITE)
+                release.stageValidation(quay_url, dest_release_tag, -1, params.PERMIT_PAYLOAD_OVERWRITE, params.PERMIT_ALL_ADVISORY_STATES, params.FROM_RELEASE_TAG, arch)
```
Comment (sosiouxme) [inline jobs/build/pre-release/Jenkinsfile:139]:
```suggestion
                release.stageValidation(quay_url, dest_release_tag, -1, params.PERMIT_PAYLOAD_OVERWRITE,  false, params.FROM_RELEASE_TAG, arch)
```
We don't actually have that param to pass in.

**Review by sosiouxme — APPROVED** [review]
LGTM with minor modification.

File: pipeline-scripts/release.groovy [hunk 2]
```diff
@@ -45,7 +45,7 @@ def destReleaseTag(String releaseName, String arch) {
  *      Valid advisories must be in QE state and have a live ID so we can
  *      include in release metadata the URL where it will be published.
  */
-Map stageValidation(String quay_url, String dest_release_tag, int advisory = 0, boolean permitPayloadOverwrite = false, boolean permitAnyAdvisoryState = false) {
+Map stageValidation(String quay_url, String dest_release_tag, int advisory = 0, boolean permitPayloadOverwrite = false, boolean permitAnyAdvisoryState = false, String nightly, String arch) {
```
Comment (sosiouxme) [inline pipeline-scripts/release.groovy:48]: ... positional parameters added after "optional" parameters? might be tripping up syntax... @joepvd any idea?

**Review by sosiouxme — CHANGES_REQUESTED** [review]
I rescind my approval... [it doesn't work](https://saml.buildvm.openshift.eng.bos.redhat.com:8888/job/hack/job/lmeyer/job/lmeyer-dev/job/dev-build%252Fpre-release/10/console).

```
org.jenkinsci.plugins.workflow.cps.CpsCompilationErrorsException: startup failed:
Script1.groovy: 125: expecting '}', found 'res' @ line 125, column 9.
           res = commonlib.shell(
           ^
```
This is from the `release.groovy` file but I can't figure out what's wrong, unless it's...

File: pipeline-scripts/release.groovy (hunk 2 above)
Comment (Ximinhan) [reply pipeline-scripts/release.groovy:48]: I reformat the indentation,don't know if it related, please check again.

**Pull request closed without merging**